### DISCUSSION TITLE
Updating a join table

### DISCUSSION THREAD

**Discussion by royaljeffr** [Budibase/budibase#4408] [Help] [answered]
I have a small app that I am trying to create  with Titles of employees and then have a list of all the attributes they could have - security controls, email lists, etc..  I have created these three tables with their fields in Postgres.
**Titles** -- id(pk), title
**Titles_Attributes** -- id (pk), description
**Titles_Attributes_Join** -- id (pk), title_id, title_attribute_id

Then in Budibase I created a Define Relationship
![image](https://user-images.githubusercontent.com/49763873/153239618-c51dcb79-6d5a-41b5-a699-ce7b543034b3.png)

On the Page I created I have a Form that displays the Title.Title and a dropdown that shows the Titles_Attributes list with checkboxes next to the attributes associated with the Titles (that data was imported at the SQL level to get started).  The problem is the join table is not updating when things are added or removed. 
![image](https://user-images.githubusercontent.com/49763873/153271427-d4cfaae4-1cbd-4639-a682-5ca351bd7a47.png)

I have tried adding two Save Row events - one pointing to each data source but I get a red toaster error with not much reason why it failed.  I even created a custom UPDATE Query but that only adds new selections, it doesn't remove the old ones.  

Any thoughts?

**Comment by mjashanks** [comment]
Hi @royaljeffr - I just attempted this, and all worked ok for me. I had a table structure just like yours.

You should not need an additional `Save Row`. So long as your relationships are setup and you are using a relationship picker then it should work. By your screenshots, It looks like everything is setup ok.

Happy to take a further look at this, if you want to export your app and send to support@budibase.com

**Comment by royaljeffr** [comment, marked as answer]
I think the issue was either a bad constraint on the database or maybe the join table had some orphaned join records that caused the app to crash.  After I deleted all the join data things are working ok
